app/services/proofing/resolution/progressive_proofer.rb

```diff
@@ -86,20 +96,80 @@ def proof_with_threatmetrix_if_needed(
         end
       end
 
-      def proof_resolution(applicant_pii:, timer:)
+      def proof_residential_address_if_needed(
+        applicant_pii:,
+        timer:,
+        double_address_verification:
+      )
+        return residential_address_unnecessary_result unless double_address_verification
+
+        timer.time('residential address') do
+          resolution_proofer.proof(applicant_pii)
+        end
+      end
+
+      def residential_address_unnecessary_result
```

Comment:
is this for the case where a user has not been presented DAV and so they only have their state id address?

Reply:
i think it might be helpful to include a short comment above each of these functions that briefly describes the use case